virttest/utils_test/__init__.py

```diff
@@ -1808,40 +1782,146 @@ def load_stress_tool(self):
+    def install(self):
+        """
+        To download, abstract, build and install the stress
+
+        :return: True if installation success, otherwise False
+        """
+        if self.check_installed():
+            error_context.context('stress has already been installed.')
+            return True
+        else:
+            error_context.context('stress has not been installed, '
+                                  'download and install...')
+            self.download_and_uncompress_stress()
             session = self.get_session()
-            session.sendline(self.stop_cmd)
-            if not self.app_running():
-                return True
-            return False
+            session.cmd_output_safe('cd %s' % self.dst_path)
+            make_cmd = self.params.get(
+                'make_cmd', './configure && make && make install')
+            error_context.context('make and install the stress app', logging.info)
+            session.cmd_output_safe(make_cmd)
+            session.close()
+            return self.check_installed()
 
-        error_context.context("stop stress app in guest", logging.info)
-        utils_misc.wait_for(_unload_stress, first=2.0,
-                            text="wait stress app quit", step=1.0, timeout=60)
+    def check_installed(self):
+        """
+        Check if stress is installed
 
-    def app_running(self):
+        :return: True if it is installed, otherwise False
         """
-        check whether app really run in background
+        check_installed_cmd = self.params.get(
+            'check_installed_cmd', 'stress --help')
+        session = self.get_session()
+        installed = session.cmd_status(check_installed_cmd) == 0
+        session.close()
+        return installed
+
+    @error_context.context_aware
+    def uninstall(self):
+        """
+        Uninstall stress application, and clean the source files
+
+        :return: True if uninstall success, otherwise False
         """
         session = self.get_session()
-        status = session.cmd_status(self.check_cmd, timeout=60)
-        return status == 0
+        if session.cmd_status('rpm -e stress') == 0:
+            session.close()
+            return True
+
+        if not self.downloaded:
+            self.download_and_uncompress_stress()
+        session.cmd_output_safe('cd %s' % self.dst_path)
+
+        uninstall_cmd = './configure && make uninstall'
+        error_context.context('Uninstall stress', logging.info)
+        status, output = session.cmd_status_output(uninstall_cmd)
+        if status != 0:
+            error_context.context('Uninstall stress failed with '
+                                  'error: %s' % output, logging.error)
+            session.close()
+            return False
+
+        error_context.context('Uninstall stress success, now '
+                              'remove the source files', logging.info)
+        rm_cmd = 'cd && rm -rf %s' % self.dst_path
```

Comment:
Do we need to remove tarball_path as well?

Reply:
Why add `cd` in this command?